receiver/osqueryreceiver/osquery.go

```diff
@@ -59,12 +67,36 @@ func newLog(ld plog.Logs, query string, row map[string]string) plog.Logs {
 	return ld
 }
 
+func (or *osQueryReceiver) connect(retries int) (*osquery.ExtensionManagerClient, error) {
+	client, err := or.config.getOsQueryClient()
+	if err != nil && retries > 0 {
+		or.logger.Info("Error connecting to osquery socket, retrying", zap.Error(err))
+		return or.connect(retries - 1)
```

Comment:
i would prefer loop instead of recursion just for the sake of readability and stack optimization